SparkBuilderGeneratorPlugin/test/com/helospark/spark/builder/handlers/it/BuilderWithSuperSetterE2ETest.java

```diff
@@ -0,0 +1,202 @@
+package com.helospark.spark.builder.handlers.it;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.any;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.dom.TypeDeclaration;
+import org.mockito.Mock;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import com.helospark.spark.builder.handlers.GenerateRegularBuilderHandler;
+import com.helospark.spark.builder.handlers.GenerateStagedBuilderHandler;
+import com.helospark.spark.builder.handlers.it.dummyService.NoDialogOperationPerformedStagedBuilderDialogAnswerProvider;
+import com.helospark.spark.builder.handlers.it.dummyService.TypeExtractorAnswerProvider;
+
+public class BuilderWithSuperSetterE2ETest extends BaseBuilderGeneratorIT {
+    private NoDialogOperationPerformedStagedBuilderDialogAnswerProvider dialogAnswerProvider = new NoDialogOperationPerformedStagedBuilderDialogAnswerProvider();
+    @Mock
+    private IType firstSuperClassType;
+    @Mock
+    private IType secondSuperClassType;
+    @Mock
+    private ICompilationUnit firstSuperClassICompilationUnit;
+    @Mock
+    private ICompilationUnit secondSuperClassICompilationUnit;
+
+    private TypeExtractorAnswerProvider typeExtractorAnswerProvider;
+
+    @BeforeMethod
+    public void beforeMethod() throws JavaModelException {
+        super.init();
+        underTest = new GenerateRegularBuilderHandler();
+
+        // first super class
+        given(firstSuperClassType.getCompilationUnit()).willReturn(firstSuperClassICompilationUnit);
+        given(firstSuperClassType.getElementName()).willReturn("TestSuperClass");
+
+        // second super class
+        given(secondSuperClassType.getCompilationUnit()).willReturn(secondSuperClassICompilationUnit);
+        given(secondSuperClassType.getElementName()).willReturn("TestSuperSuperClass");
+
+        // When getting the superClass return the correct type
+        Map<String, IType> map = new HashMap<>();
```

Comment:
Java 9?

Reply:
I don't want to increase the minimum JRE requirement for the plugin (and Eclipse).

Reply:
I totally understand your point.